packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -724,30 +720,195 @@ class TimeInput extends React.Component {
 
-    const formatDescriptionId = `terra-time-input-description-format-${this.uuid}`;
-
-    const format = showSeconds
-      ? `(${intl.formatMessage({ id: 'Terra.timeInput.hh' })}:${intl.formatMessage({ id: 'Terra.timeInput.mm' })}:${intl.formatMessage({ id: 'Terra.timeInput.ss' })})`
-      : `(${intl.formatMessage({ id: 'Terra.timeInput.hh' })}:${intl.formatMessage({ id: 'Terra.timeInput.mm' })})`;
+    /**
+     * Get the mask string to place under the time input fields.
+     * @returns {String} a visual hint of the time format, like 'hh:mm'.
+     */
+    function mask() {
+      if (showSeconds) {
+        return intl.formatMessage({
+          id: 'Terra.timeInput.maskHourMinute',
+          defaultMessage: '(hh:mm:ss)',
+          description: 'A visual hint that the Time Input expects two-digit hour, minute and second.',
+        });
+      }
+      return intl.formatMessage({
+        id: 'Terra.timeInput.maskHourMinuteSecond',
+        defaultMessage: '(hh:mm)',
+        description: 'Like Terra.timeInput.maskHourMinute but no second.',
+      });
+    }
+
+    /**
+     * Get the description for the hour field.
+     *
+     * NOTE: the description does not take the place of validation or error messages.
+     * @returns {String} a description of the hour field, intended to be read to screen reader users.
+     */
+    function hourDescription() {
+      if (is12Hour) {
+        return intl.formatMessage({
+          id: 'Terra.timeInput.hourDescriptionTwelve',
+          defaultMessage: 'A two-digit 12-hour value',
+          description: `Explains to screen reader users that the hour field needs a two digit hour. This will be read
+          only when a screen reader is enabled. It is never displayed. It will be read when the user has focused on
+          the hour input or when the screen reader is reading the page to the user.`,
+        });
+      }
+      return intl.formatMessage({
+        id: 'Terra.timeInput.hourDescriptionTwentyFour',
+        defaultMessage: 'A two-digit 24-hour value',
+        description: 'Like Terra.timeInput.hourDescriptionTwelve but for a 24-hour clock.',
+      });
+    }
+
+    // Fan out some component-level props into input-level a11y attributes. See the big comment below for more info.
+    inputAttributes.isInvalid = isInvalid;
+    inputAttributes.disabled = disabled;
+    inputAttributes.required = required;
+
+    const a11yString = TimeUtil.getA11YTimeValue(this.props, this.state, this.postMeridiem);
+
+    function groupLabel() {
+      if (label) {
+        return intl.formatMessage({ // fix
+          id: 'Terra.timeInput.inputGroupValue',
+          defaultMessage: '{label}',
+          description: `This will be read by screen readers as the reader moves into the group of inputs. It is intended to
+      help the user understand "you are about to enter a section of the page where many different inputs all work
+      together for this one concept of time.`,
+        }, { label });
```

Comment:
I would think we should just use the provided label as is without having to do any kind of formatting/translation.

Reply:
Could be, but I'm not sure if in another language there would be other words given?

Reply:
Thinking about this more, I've convinced myself it's OK. Will do.

Reply:
The expectation should be that the higher order component would already have translated the label that is passed to time-input. It's similar to other components (e.g. [terra-form-field](https://github.com/cerner/terra-core/blob/terra-form-fieldset%402.63.2/packages/terra-form-field/src/Field.jsx#L182)) that would just use the passed in label as is.